src/BigIntMsg.chpl

```diff
@@ -68,22 +68,21 @@ module BigIntMsg {
                 if && reduce (tmp == 0) {
                   // early out if we are already all zeroes
                   var retname = st.nextName();
-                  st.addEntry(retname, new shared SymEntry(tmp:uint));
+                  st.addEntry(retname, new shared SymEntry(makeDistArray(tmp.size, uint)));
                   retList.append("created %s".format(st.attrib(retname)));
                 }
                 else {
+                  var low: [tmp.domain] uint;
                   while || reduce (tmp!=0) {
-                    var low: [tmp.domain] bigint;
-                    // create local copy, needed to work around bug fixed in Chapel, but
-                    // needed for backwards compatability for now
-                    forall (lowVal, tmpVal) in zip(low, tmp) with (var local_block_size = block_size) {
-                        lowVal = tmpVal % local_block_size;
-                    }
+                    low = tmp:uint;
                     var retname = st.nextName();
 
-                    st.addEntry(retname, new shared SymEntry(low:uint));
+                    st.addEntry(retname, new shared SymEntry(low));
                     retList.append("created %s".format(st.attrib(retname)));
-                    tmp /= block_size;
+                    forall t in tmp with (var local_block_size = block_size) {
+                      // TODO eventually we will likely want to >>= 64 here
+                      t /= local_block_size;
+                    }
```

Comment:
All this looks good. There is almost certainly more we could do somewhat easily (`>>=`, check `tmp != 0` as part of the forall) and even more we can do with more work (directly extract limbs) but this all seems like a really good step forward.

Reply:
i think I avoided the `>>=` due to the [right shift bug](https://github.com/chapel-lang/chapel/issues/21206). I don't think that's in our lowest supported version. But now that I think about it, i could just do `tmp >>= 64:uint` right?

I really appreciate the feedback! Looking at how you sped up the `bigint_from_uint_arrays` case was helpful when looking for low hanging fruit. I'm super glad to see these performance increases that will hopefully make this a viable solution for our users in the short term!

Reply:
Ah, that makes sense. Yeah, I think `>>= 64:uint` would work.

Reply:
@ronawho i implemented a few of these changes (`>>=` and `tmp != 0` in the forall) and see a half second speedup.  It also makes the code cleaner in my opinion. If you wouldn't mind checking it out and verifying it makes sense, that would be great!

Reply:
👍 looks good, and shows additional speedup for multi-locale compared to the original PR